c2_components/include/mfx_c2_encoder_component.h

```diff
@@ -220,6 +220,13 @@ class MfxC2EncoderComponent : public MfxC2Component
     // Input frame info with width or height not 16byte aligned
     mfxFrameInfo m_mfxInputInfo;
 
+#if MFX_DEBUG_DUMP_FRAME_ENC == MFX_DEBUG_YES
+    int m_count = 0;
+    std::mutex m_count_lock;
+    FILE* m_file = 0;
+    bool NeedDumpBuffer();
+#endif
```

Comment:
they are only used for dump function, then it is not needed placed under class MfxC2EncoderComponent, we can keep member of class minimal